lib/Runtime/Library/WebAssembly.cpp

```diff
@@ -34,9 +34,9 @@ Var WebAssembly::EntryCompile(RecyclableObject* function, CallInfo callInfo, ...
         WebAssemblySource src(args[1], true, scriptContext);
         module = WebAssemblyModule::CreateModule(scriptContext, &src);
     }
-    catch (JavascriptError & e)
+    catch (JavascriptException & e)
```

Comment:
I don't know how I got the initial implementation... it simply doesn't work wince we don't throw JavascriptErrors, we throw JavascriptException.
This caused to throw when calling WebAssembly.instantiate/compile instead of returning a rejecting Promise